source/_integrations/sma.markdown

```diff
@@ -4,134 +4,67 @@ description: Instructions on how to connect your SMA Solar Inverter to Home Assi
-| frequency  | Hz | Grid frequency. |
-| voltage_l1 | V | Voltage for phase 1. |
-| voltage_l2 | V | Voltage for phase 2. |
-| voltage_l2 | V | Voltage for phase 3. |
-| status     |  | Status of the solar plant. |
+| grid_power   | W    | Power supplied by the grid.   |
+| frequency    | Hz   | Grid frequency.   |
+| voltage_l1   | V    | Voltage for phase 1.   |
+| voltage_l2   | V    | Voltage for phase 2.   |
+| voltage_l3   | V    | Voltage for phase 3.   |
+| current_l1   | A    | Current for phase 1.   |
+| current_l2   | A    | Current for phase 2.   |
+| current_l3   | A    | Current for phase 3.   |
+| power_l1     | W    | Power for phase 1.   |
+| power_l2     | W    | Power for phase 2.   |
+| power_l3     | W    | Power for phase 3.   |
+| pv_power     | W    | Current power generated by the inverter. (Should give same value as grid_power)   |
+| pv_power_a   | W    | Current power generated by the solar panels (A side).   |
+| pv_power_b   | W    | Current power generated by the solar panels (B side).    |
+| pv_voltage_a | V    | Current voltage generated by the solar panels (A side).   |
+| pv_voltage_b | V    | Current voltage generated by the solar panels (B side).   |
+| pv_current_a | A    | Current amperage generated by the solar panels (A side).   |
+| pv_current_b | A    | Current amperage generated by the solar panels (B side).   |
+| pv_gen_meter | kWh  | Total kWh generated to date.   |
+| total_yield  | kWh  | Total power yield from a solar installation.   |
+| daily_yield  | Wh   | The solar plant's yield for today   |
+| grid_power_supplied | W    | Power supplied.   |
+| grid_power_absorbed | W    | Power absorbed.   |
+| grid_total_yield | kWh  | Total power from the grid.   |
+| grid_total_absorbed | kWh  | Total power supplied to the grid.   |
+| current_consumption | W    | Power that you are currently drawing, depending on your installation it can be a combination of the inverter and the grid.   |
+| total_consumption | kWh  | Total power consumption.   |
+
+### Custom Sensors
 
-The SMA WebConnect module supports a wide variety of sensors, and not all these have been mapped in the `pysma` library. Custom sensors can be defined by using the `custom` section of the configuration. You will need: A sensor name (no spaces), the SMA sensor key and the unit.
+The SMA WebConnect module supports a wide variety of sensors, and not all these have been mapped in the `pysma` library. Custom sensors can be defined by selecting the `add custom sensor` checkbox during the setup and entering the required information in the next prompt. You will need: A sensor name, the SMA sensor key and the unit. The path attribute is a custom JMESpath to extract the sensor value and typically not required (advanced use only).
 
 Example:
 
 ```yaml
-   custom:
-      yesterday_consumption:
-         key: "6400_00543A01"
-         unit: kWh
-         factor: 1000
+name: yesterday_consumption
+key: 6400_00543A01
+unit: kWh
+factor: 1000
 ```
```

Comment:
This is not really a documentation comment, but since I'm here I'm going to place it anyway.

I feel like this is not a feature that should go into the UI. In fact, I think this is a feature that doesn't belong in Home Assistant at all. `pysma` already has a mapping for a lot of sensor, they should be extended over time. Additionally, from the Home Assistant end of things, we should only enable the most common sensors by default (and have the others disabled by default).

Reply:
Thanks for the feedback @frenck! I have implemented a config flow that supports all the config options that are now available in the yaml file. I agree that preferably no custom sensor config would be needed and all sensors are mapped in the `pysma` library, but removing it completely would be a breaking change for users that actually use them.

@kellerza what is your experience with the custom sensors? Do you know if they are still used a lot?

Reply:
The sensors are different in every other SMA inverter, so always got the feeling there are widely used from community discussions. 

Will it be possible to keep the yaml config for custom sensors to augment the sensors available to the UI? Especially for testing this would be great. (Although this probably brought us to this specific problem in the first place)

On the other hand, this would force people to update sensors values in the library and we can have an inverter name there as well

Reply:
I could remove the UI for setting custom sensors. Creating  custom sensors would still be possible via the yaml import, but deprecated. The custom sensors will (probably) keep working for existing users and if we remove the documentation new users are more likely to update sensors values in the library.

Reply:
That could work yes. Lets keep custom_sensors in yaml as a way to test custom sensors with a warning that it should be worked into the pysma lib. 

We can work the existing custom sensor docs into pysma as a section on how to test& PR new sensors

Reply:
I have removed the custom sensor flow step and updated the documentation with a yaml example to test custom sensors.